rust/agama-locale-data/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+use serde::Deserialize;
+use std::fs::File;
+use std::io::BufReader;
+use quick_xml::de::Deserializer;
+use flate2::bufread::GzDecoder;
+
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RankedLanguage {
+    #[serde(rename(deserialize = "languageId"))]
+    pub id: String,
+    pub rank: u16
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RankedLanguages {
+    pub language: Option<Vec<RankedLanguage>>
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RankedTerritory {
+    #[serde(rename(deserialize = "territoryId"))]
+    pub id: String,
+    pub rank: u16
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RankedTerritories {
+    pub territory: Option<Vec<RankedTerritory>>
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Keyboard {
+    #[serde(rename(deserialize = "keyboardId"))]
+    pub id: String,
+    pub description: String,
+    pub ascii: bool,
+    pub comment: Option<String>,
+    pub languages: RankedLanguages,
+    pub territories: RankedTerritories
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Keyboards {
+    pub keyboard: Vec<Keyboard>
```

Comment:
np: If you do not plan to add anything else, you could use the [newtype pattern idiom](https://doc.rust-lang.org/rust-by-example/generics/new_types.html).